collect_app/src/main/java/org/odk/collect/android/workers/AutoSendWorker.java

```diff
@@ -18,79 +18,133 @@
 import org.odk.collect.android.application.Collect;
 import org.odk.collect.android.dao.FormsDao;
 import org.odk.collect.android.dao.InstancesDao;
+import org.odk.collect.android.dto.Form;
+import org.odk.collect.android.utilities.IconUtils;
+import org.odk.collect.android.utilities.gdrive.GoogleAccountsManager;
+import org.odk.collect.android.utilities.InstanceUploaderUtils;
 import org.odk.collect.android.listeners.InstanceUploaderListener;
 import org.odk.collect.android.preferences.GeneralSharedPreferences;
 import org.odk.collect.android.preferences.PreferenceKeys;
 import org.odk.collect.android.provider.InstanceProviderAPI.InstanceColumns;
 import org.odk.collect.android.tasks.InstanceGoogleSheetsUploader;
 import org.odk.collect.android.tasks.InstanceServerUploader;
-import org.odk.collect.android.tasks.ServerPollingJob;
-import org.odk.collect.android.utilities.IconUtils;
-import org.odk.collect.android.utilities.InstanceUploaderUtils;
 import org.odk.collect.android.utilities.PermissionUtils;
-import org.odk.collect.android.utilities.gdrive.GoogleAccountsManager;
 
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+
+import androidx.work.Worker;
+import timber.log.Timber;
 
 import static org.odk.collect.android.provider.FormsProviderAPI.FormsColumns.AUTO_SEND;
 import static org.odk.collect.android.utilities.ApplicationConstants.RequestCodes.FORMS_UPLOADED_NOTIFICATION;
 
-public class AutoSendWorker extends BroadcastReceiver implements InstanceUploaderListener {
-
-    // turning on wifi often gets two CONNECTED events. we only want to run one thread at a time
-    public static boolean running;
+public class AutoSendWorker extends Worker implements InstanceUploaderListener {
     InstanceServerUploader instanceServerUploader;
 
     InstanceGoogleSheetsUploader instanceGoogleSheetsUploader;
 
     private String resultMessage;
+    /**
+     * Causes the doWork method to wait until the instanceUploader AsyncTask has completed.
```

Comment:
Why can't the doWork method do the work synchronously?

Reply:
It absolutely should -- that's what I'm getting at about the need to restructure the uploaders. The code for uploading in the auto-send/auto send/autoSend case is the same as the code for manually uploading which is done by an `AsyncTask` currently. One option would be to pull all that out into a separate class that both the worker and the asynctask use. I feel like there's probably a better structure to be had, though.